server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java

```diff
@@ -143,4 +213,13 @@ private String convertCategory(Constants.EventCategory category) {
   public String getDescription() {
     return "Persist component links";
   }
+
+  private enum EventDtoToHash implements Function<EventDto, Integer> {
+    INSTANCE;
+
+    @Override
+    public Integer apply(EventDto input) {
+      return Objects.hashCode(input.getComponentUuid(), input.getSnapshotId(), input.getDate(), input.getCategory(), input.getName(), input.getDescription(), input.getData());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) input must be non-null but is marked as nullable [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE)